src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -163,8 +163,16 @@ protected override void LoadContext(ExecutionContext context)
             base.LoadContext(context);
         }
 
-        internal void LoadContext(ExecutionContext context, bool checkReturnValue)
+        internal protected void LoadClonedContext(ExecutionContext context, int initialPosition, bool checkReturnValue)
         {
+            // Copy script hash
+            var state = context.GetState<ExecutionContextState>();
+            context = context.Clone(initialPosition);
+            var newState = context.GetState<ExecutionContextState>();
+            newState.CallFlags = state.CallFlags;
+            newState.ScriptHash = state.ScriptHash;
+            newState.CallingScriptHash = state.CallingScriptHash;
```

Comment:
@erikzhang Should we use the cloned `CallingScriptHash` or the current one (`state.ScriptHash`)?

Reply:
Where is it used? I think we only need to modify `LoadContract`.

Reply:
We need to copy the script hash, or it will be wrong computed

Reply:
The state is shared with every cloned contexts.

https://github.com/neo-project/neo-vm/blob/22ff9d6712166a8aab8f1baf26734b9e57876910/src/neo-vm/ExecutionContext.cs#L11

Reply:
Fixed